src/dataflow/components/dataflow-program.tsx

```diff
@@ -48,42 +52,42 @@ export class DataflowProgram extends BaseComponent<IProps, IState> {
         new RelayReteNodeFactory(numSocket)];
       if (!this.toolDiv) return;
```

Comment:
`componentDidMount()` is only called once, so if it is possible to get here without a valid `this.toolDiv` then bailing at this point means that we'll never complete the rest of the initialization. I think the most straightforward way to deal with this would be to move the initialization into a separate function and then in `componentDidMount()` and `componentDidUpdate()` have code something like:
```typescript
  if (!this.programEditor && this.toolDiv) {
    this.initProgramEditor();
  }
```

Reply:
Fixed.